spatialmath/box.go

```diff
@@ -352,3 +355,84 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// TODO: add function description
+func (b *box) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+
+	dims := [3]float64{b.halfSize[0] * 2, b.halfSize[1] * 2, b.halfSize[2] * 2}
+
+	var faces [][]float64
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += options["xIter"].(float64) {
+		for k := 0.0; k <= dims[2]; k += options["zIter"].(float64) {
+			p1 := []float64{i, 0, k}
+			p2 := []float64{i, dims[1], k}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// which faces are these
+	for j := 0.0; j <= dims[1]; j += options["yIter"].(float64) {
```

Comment:
This should probably be a helper function

Reply:
agreed

Reply:
will make these three for loops concurrent